File: Programers/Level2/모음사전.py
def solution(word):
    answer = 0
    
    aeiou = ['A','E','I','O','U']
    visited = [False] * 5
    arr = []
    def dfs(st, temp, idx) :
        
        if idx == 6 :
            return
        
        
        if st != '' :
            arr.append(st)
            
        for i in range(5) :
            dfs(st + temp[i], temp, idx+1)
          
    dfs('', aeiou, 0)
    
    answer = arr.index(word) +1
    
    return answer

File: Javascript/Programmers/모음사전.js
function solution(word) {
    var answer = 0;
    
    alpha = ['A','E','I','O','U'];
    arr = [];
    
    function dfs(st, alpha, idx) {
        
        if (idx == 6){
            return;
        }
        if (st != ''){
            arr.push(st);
        }
        for(let i = 0; i < alpha.length; i++){
            dfs(st + alpha[i], alpha, idx + 1);
        }
    }
    
    dfs('', alpha, 0);    
    answer = arr.indexOf(word) + 1;
    return answer;
}
